Sources/PrimerSDK/Classes/Core/BIN Data/CardValidationService.swift

```diff
@@ -0,0 +1,234 @@
+//
+//  CardValidationService.swift
+//  PrimerSDK
+//
+//  Created by Jack Newcombe on 30/10/2023.
+//
+
+import Foundation
+
+protocol CardValidationService {
+    static var apiClient: PrimerAPIClientProtocol? { get set }
+    func validateCardNetworks(withCardNumber cardNumber: String)
+}
+
+class DefaultCardValidationService: CardValidationService, LogReporter {
+    
+    static let maximumBinLength = 8
+    
+    static var apiClient: PrimerAPIClientProtocol?
+
+    var delegate: PrimerHeadlessUniversalCheckoutRawDataManagerDelegate? {
+        return self.rawDataManager.delegate
+    }
+        
+    let apiClient: PrimerAPIClientBINDataProtocol
+    
+    let debouncer: Debouncer
+    
+    let rawDataManager: PrimerHeadlessUniversalCheckout.RawDataManager
+    
+    let allowedCardNetworks: [CardNetwork]
+
+    var mostRecentCardNumber: String?
+
```

Comment:
Could we review the above for access control?
A quick look says `maximumBinLength ` `debouncer` and `mostRecentCardNumber` dont need to be accessible outside the class, might be others

Reply:
Ah these were used in the tests at one point but aren't any more - will update 👍 

The exception is `maximumBinLength` - this probably should be used in the tests in place of magic numbers, so will update them too 👍